agent/node.go

```diff
@@ -627,7 +627,19 @@ func (n *Node) runManager(ctx context.Context, securityConfig *ca.SecurityConfig
 			ElectionTick:   n.config.ElectionTick,
 		})
 		if err != nil {
-			return err
+			logrus.Errorf("error starting manager: %v", err)
+			// Something went wrong, but it might be fixed later on
+			// by user based on the error log. So rather than quit,
+			// we keep waiting for my role to be changed back if
+			// promotion failed. On the meantime, the agent part still
+			// work.
+			// TODO(jinuxstyle): change role back to agent if promotion
+			// failed in a self-healing way.
+			n.waitRole(ctx, ca.AgentRole)
+			if ctx.Err() != nil {
```

Comment:
In practice, it probably is, but that should not be relied upon. The interface indicates that `ctx.Done()` should be closed before the value is valid.